File: apps/backend/src/EnterpriseAgentOs.Domain/features/Agents/IAgentService.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IAgentService
{
    Task<IReadOnlyList<AgentDto>> ListAsync(CancellationToken ct = default);
    Task<AgentDto?> GetAsync(Guid id, CancellationToken ct = default);
    Task<AgentDto> CreateAsync(CreateAgentRequest request, Guid? ownerId = null, CancellationToken ct = default);
    Task<AgentDto?> PatchAsync(Guid id, PatchAgentRequest request, CancellationToken ct = default);
    Task<bool> DeleteAsync(Guid id, CancellationToken ct = default);
    Task InitializeAgentAsync(Guid agentId, Guid userId, AgentInitRequest init, CancellationToken ct = default);
}

public sealed record PatchAgentRequest(string? Provider, string? Model, string? Name = null, string? Prompt = null);

public sealed record AgentInitRequest(
    IReadOnlyList<string>? ToolNames,
    IReadOnlyList<AgentToolPermissionInit>? ToolPermissions,
    IReadOnlyList<string>? ChannelSlugs,
    string? BootstrapMessage);

public sealed record AgentToolPermissionInit(string Tool, ToolPermission Mode);

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentService.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.dtos.AgentDto;
import com.enterpriseagentos.backendjava.domain.features.agents.dtos.AgentInitRequest;
import com.enterpriseagentos.backendjava.domain.features.agents.dtos.CreateAgentRequest;
import com.enterpriseagentos.backendjava.domain.features.agents.dtos.PatchAgentRequest;

public interface IAgentService {
    CompletableFuture<List<AgentDto>> listAsync();
    CompletableFuture<AgentDto> getAsync(UUID id);
    CompletableFuture<AgentDto> createAsync(CreateAgentRequest request, UUID ownerId);
    CompletableFuture<AgentDto> patchAsync(UUID id, PatchAgentRequest request);
    CompletableFuture<Boolean> deleteAsync(UUID id);
    CompletableFuture<Void> initializeAgentAsync(UUID agentId, UUID userId, AgentInitRequest init);
}
